src/runtime/join_handle.rs

```diff
@@ -23,28 +21,20 @@ pub(crate) enum AsyncJoinHandle<T> {
 }
 
 impl<T> Future for AsyncJoinHandle<T> {
-    // tokio wraps the Output of its JoinHandle in a Result, while async-std does not.
-    // In order to standardize on this without us panicking, we too wrap the output in a Result.
-    type Output = Result<T>;
+    // tokio wraps the Output of its JoinHandle in a Result that conatins an error if the task
+    // panicked, while async-std does not.
+    //
+    // Given that async-std will panic or abort the task in this scenario, there is not a
```

Comment:
Was this change made to help with debugging?

Reply:
Were we ever returning an error to user that we're now panicking on, or is this just internal?

Reply:
Nope, we only ever actually used the join handle in the CMAP spec tests, and we were always unwrapping them there anyways. The driver mostly used `execute` before this PR which discarded the result of the async task.